scipy/optimize/tests/test_minpack.py

```diff
@@ -550,6 +550,11 @@ def test_function_zero_params(self):
         # Fit args is zero, so "Unable to determine number of fit parameters."
         assert_raises(ValueError, curve_fit, lambda x: x, [1, 2], [3, 4])
 
+    def test_None_x(self):
+        popt, pcov = curve_fit(lambda _, a: a * np.arange(10),
```

Comment:
Could you add a comment:
`# Added in GH10196`
?

Reply:
done